src/Microsoft.Management.Deployment/Helpers.cpp

```diff
@@ -135,15 +137,18 @@ namespace winrt::Microsoft::Management::Deployment::implementation
             UINT32 length = ARRAYSIZE(packageFamilyName);
             if (::GetPackageFamilyName(processHandle.get(), &length, packageFamilyName) == ERROR_SUCCESS)
             {
-                return { packageFamilyName };
+                // If the package is calling into itself, fall through to the executable name
+                if (AppInstaller::Runtime::GetPackageFamilyName() != packageFamilyName)
+                {
+                    return { packageFamilyName };
+                }
             }
 
             // if the caller doesn't have an AppUserModelID then fall back to the executable name
-            wil::unique_cotaskmem_string imageName = nullptr;
-            if (SUCCEEDED(wil::QueryFullProcessImageNameW(processHandle.get(), 0, imageName)) &&
-                (imageName.get() != nullptr))
+            std::filesystem::path executablePath = AppInstaller::Filesystem::GetExecutablePathForProcess(processHandle.get());
+            if (executablePath.has_filename())

```

Comment:
If the executable path does not have file name, should we return the full path instead of empty?

Reply:
I don't think so; that sounds like an error and logging would be appropriate.